internal/alert/telegram.go

```diff
@@ -0,0 +1,49 @@
+package alert
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/ianmarmour/nvidia-clerk/internal/config"
+)
+
+type TelegramPayload struct {
+	ChatID string `json:"chat_id"`
+	Text   string `json:"text"`
+}
+
+//SendTelegramMessage Sends a notification message to a Telegram Webhook.
+func SendTelegramMessage(item string, config config.TelegramConfig, client *http.Client) error {
+	content := fmt.Sprintf("%s Ready for Purchase", item)
+
+	payload, err := json.Marshal(TelegramPayload{fmt.Sprintf("@%s", config.ChatID), content})
+
+	if err != nil {
+		log.Fatalln("Unable to marshal telegram payload.")
+	}
+
+	req, _ := http.NewRequest("POST", fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", config.APIKey), bytes.NewBuffer(payload))
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, _ := client.Do(req)
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+		bodyString := string(bodyBytes)
+
+		fmt.Println(bodyString)
+		fmt.Println("Unable to send message to telegram, bad request.")
```

Comment:
Prefer log.Println (it has timestamping)

Reply:
Agreed, I blame that on rushing because I'm at work. That entire section of code was a test and has been removed.